packages/cli/src/metadataGeneration/typeResolver.ts

```diff
@@ -161,14 +161,18 @@ export class TypeResolver {
           const propertyType = typeChecker.getTypeOfSymbolAtLocation(property, this.typeNode);
           const declaration = getDeclaration(property) as ts.PropertySignature | ts.PropertyDeclaration | ts.ParameterDeclaration | undefined;
 
-          if (declaration && ts.isPropertySignature(declaration)) {
-            return this.propertyFromSignature(declaration, mappedTypeNode.questionToken);
-          } else if (declaration && (ts.isPropertyDeclaration(declaration) || ts.isParameter(declaration))) {
-            return this.propertyFromDeclaration(declaration, mappedTypeNode.questionToken);
+          // Note: for conditional type node, we want to resolve propertyType, because the declaration doesn't have the right type
```

Comment:
This was never a good idea, I'm just not sure what the best way to get the type of a property in a mapped type is.
Instead of piling up special cases, maybe it's a good time to see what the "proper" resolution would be.

Reply:
Do you have any leads? I haven't seen a magic way to make every cases works easily 😕  But I agree having special cases is not ideal

Reply:
I haven't found a good way to get the Type, maybe we could try an approach where we add the Type Arguments to the Context + the iterator (`key` in #824 ) and resolve the TypeNode given the Context.

Reply:
I've tried but I haven't found anything that works, I will try to take time for this later, in the meantime, if you have time, I've allowed edits on this PR so maybe you can try too